guides/developer/getting-set-up.md

```diff
@@ -0,0 +1,86 @@
+
+# System Requirements
+
+This guide assumes you have access to a machine with a minimum specification of:
+
+- At least 16 GB of RAM, ideally more
+- A 4 core CPU
+
+Spinnaker is developed on Linux. This guide uses Halyard which requires one of the following
+flavors of Linux:
+
+- Ubuntu 14.04, 16.04 or 18.04
+
+# Fetching Spinnaker's codebase
+
+The following steps will install Spinnaker's management tool, Halyard, fetch Spinnaker's
+codebase, and perform just enough configuration to get Spinnaker up and running.
+
+Each of these steps will take between 5 and 30 minutes to finish and may require you to
+do multiple things, such as installing CLI tools or configuring service accounts. A good
+way to approach this process is to open a step in its own browser tab and then work through
+it to completion, closing it (and any others you've opened in support of it) once it's done.
+
+1. [Install Halyard](/setup/install/halyard/#install-on-debianubuntu-and-macos)
+1. [Set up a storage service](/setup/install/storage/)
+1. [Set up your cloud provider of choice](/setup/install/providers)
+1. [Configure a LocalGit deployment](/setup/install/environment/#local-git)
+1. Run `hal deploy apply`
+
+## What does this do?
+
+Halyard creates a directory at `~/dev/spinnaker` with the following contents:
+- a subdirectory for each Spinnaker service containing that service's source code
+- a `scripts` directory
+- a `logs` directory
+- a set of `.pid` files, one for each service that is running
+
+The `scripts` directory contains scripts to start and stop each individual service. For
+example calling `~/dev/spinnaker/scripts/deck-stop.sh` will kill the Deck process and
+delete the `deck.pid` file. Running `~/dev/spinnaker/scripts/deck-start.sh` will restart Deck
+and recreate its `deck.pid` file.
+
+# Making Changes to Spinnaker
+
+Once you have a working LocalGit deployment you can begin to make changes to the codebase.
+After you've made edits in the code of a service you can see those changes reflected
+by restarting the service you've modified.
+
+To restart a service use the stop and start scripts in `~/dev/spinnaker/scripts`. The only
+service that does not require this kind of restart is Deck; its webserver watches for file
+changes and re-compiles the application as necessary.
+
+Follow the [contributing guidelines](https://www.spinnaker.io/community/contributing/)
+if you plan to submit your work as a patch to the open source project.
+
+# Configuring an IDE
+
+## Intellij
+
+Run `./gradlew idea` in a service's directory to generate intellij project files
+
+# Debugging
+
+Each Java service can be configured to listen for a debugger. To start the JVM in debug
+mode, set the Java system property DEBUG=true:
+
+```
+./gradlew -DDEBUG=true
+```
+
+The JVM will then listen for a debugger to be attached on a port specific to that service. You can
+discover which port by checking the service's README.md file in your local checkout of its git repo.
```

Comment:
Might be helpful to list the debug-listening port numbers here too, similarly to how we list the respective port mappings [here](https://www.spinnaker.io/reference/architecture/#port-mappings)? Save them the trouble of going to the readme.md for each service.

Reply:
Ah, nice.  I think I might link out to the architecture section from this doc as well.  Those ports are mighty useful.

Reply:
I added most of the debugging ports, and always picked the service port + 100.  There might be one or two that were added before that don't match that, but maybe we could just change them to make it easy.

Reply:
I've added the table of debugging ports and linked out to the architecture pages.